src/utils.cpp

```diff
@@ -226,4 +230,257 @@ namespace diskann {
+                                                    int32_t*&          data,
+                                                    size_t& npts, size_t& ndim,
+                                                    size_t offset);
+
+  template DISKANN_DLLEXPORT void copy_aligned_data_from_file<uint8_t>(
+      AlignedFileReader& reader, uint8_t*& data, size_t& npts, size_t& dim,
+      const size_t& rounded_dim, size_t offset);
+  template DISKANN_DLLEXPORT void copy_aligned_data_from_file<int8_t>(
+      AlignedFileReader& reader, int8_t*& data, size_t& npts, size_t& dim,
+      const size_t& rounded_dim, size_t offset);
+  template DISKANN_DLLEXPORT void copy_aligned_data_from_file<float>(
+      AlignedFileReader& reader, float*& data, size_t& npts, size_t& dim,
+      const size_t& rounded_dim, size_t offset);
+
+  template DISKANN_DLLEXPORT void read_array<char>(AlignedFileReader& reader,
+                                                   char* data, size_t size,
+                                                   size_t offset);
+
+  template DISKANN_DLLEXPORT void read_array<uint8_t>(AlignedFileReader& reader,
+                                                      uint8_t*           data,
+                                                      size_t             size,
+                                                      size_t offset);
+  template DISKANN_DLLEXPORT void read_array<int8_t>(AlignedFileReader& reader,
+                                                     int8_t* data, size_t size,
+                                                     size_t offset);
+  template DISKANN_DLLEXPORT void read_array<uint32_t>(
+      AlignedFileReader& reader, uint32_t* data, size_t size, size_t offset);
+  template DISKANN_DLLEXPORT void read_array<float>(AlignedFileReader& reader,
+                                                    float* data, size_t size,
+                                                    size_t offset);
+
+  template DISKANN_DLLEXPORT void read_value<uint8_t>(AlignedFileReader& reader,
+                                                      uint8_t&           value,
+                                                      size_t offset);
+  template DISKANN_DLLEXPORT void read_value<int8_t>(AlignedFileReader& reader,
+                                                     int8_t&            value,
+                                                     size_t             offset);
+  template DISKANN_DLLEXPORT void read_value<float>(AlignedFileReader& reader,
+                                                    float&             value,
+                                                    size_t             offset);
+  template DISKANN_DLLEXPORT void read_value<uint32_t>(
+      AlignedFileReader& reader, uint32_t& value, size_t offset);
+  template DISKANN_DLLEXPORT void read_value<uint64_t>(
+      AlignedFileReader& reader, uint64_t& value, size_t offset);
+
+#endif
+
 }  // namespace diskann
```

Comment:
add new line at the end. Clang-format might fix this.

Reply:
yes. fixed.